### PR TITLE
S2U-32 Assignments: add support for tags

### PR DESCRIPTION
**Pull request by bgarciaentornos**

### CODE REVIEW

File: vuecomponents/tool/src/main/frontend/src/components/tag-selector.vue [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        name: newTag,
+        code: newTag
+      }
+      this.options.push(tag);
+      if (isSelected){
+        this.value.push(tag)
+      }
+    }
+  },
+  async mounted () {console.log('eee');
```
Comment (jesusmmp) [inline vuecomponents/tool/src/main/frontend/src/components/tag-selector.vue:134]: Delete console.log

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 2]
```diff
@@ -2935,6 +2963,37 @@ private String build_list_assignments_context(VelocityPortlet portlet, Context c
...
+                    String modifier = userDirectoryService.getUser(a.getModifier()).getDisplayName();
+                    tagLabels.add(modifier);
+                } catch (Exception e) {
+                    log.warn(this + ":build_list_assignments_context cannot get user " + e.getMessage() + " user id=" + a.getModifier());
+                }
+            }
+        }
+        if (a.getProperties().get(NEW_ASSIGNMENT_TAG_GROUPS) != null && Boolean.parseBoolean(a.getProperties().get(NEW_ASSIGNMENT_TAG_GROUPS))) {
+            try {
+                Site site = siteService.getSite(a.getContext());//TODO 1 general?
```
Comment (jesusmmp) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:2986]: Delete comment

File: library/pom.xml [hunk 3]
```diff
@@ -316,6 +316,7 @@
 			<artifactId>dompurify</artifactId>
 			<version>2.3.6</version>
 		</dependency>
+
```
Comment (jesusmmp) [inline library/pom.xml:319]: Undo.

File: vuecomponents/bundle/src/main/bundle/tag-selector_es.properties [hunk 4]
```diff
@@ -0,0 +1,8 @@
+search_or_add=Buscar o añadir etiqueta
```
Comment (jesusmmp) [inline vuecomponents/bundle/src/main/bundle/tag-selector_es.properties:1]: Should we replace ñ by \u00f1 or it is OK?

Comment (bgarciaentornos) [reply vuecomponents/bundle/src/main/bundle/tag-selector_es.properties:1]: It seems to work fine this way so I would leave it... It's easier to read and maintain

File: assignment/api/src/resources/assignment_eu.properties [hunk 5]
```diff
@@ -1106,3 +1106,13 @@ autoevaluation=Autoebaluazioa:
...
+
+#S2U-32
+tags_header=Etiquetas
+tags_groups=A\u00f1adir autom\u00e1ticamente etiquetas para los grupos seleccionados.
+tags_instructor=A\u00f1adir autom\u00e1ticamente etiquetas para el autor.
+tags_note=Nota: no aparecer\u00e1n en el selector al editar
+tags_search=Buscar usando etiquetas
+tags_option_students=Mostrar etiquetas a los estudiantes.
+tags_option_students_default=Por defecto (ocultar etiquetas a los estudiantes).
+tags_option_students_yes=Mostrar etiquetas a los estudiantes y permitirles el uso del filtro.
```
Comment (bgarciaentornos) [inline assignment/api/src/resources/assignment_eu.properties:1118]: @PabloSanRoman would you mind providing translations for EU?

File: tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/TagServiceImpl.java [hunk 6]
```diff
@@ -78,15 +88,74 @@ public void destroy() {
     }
 
     @Override
-    public TagCollections getTagCollections() {
-        return tagCollections;
+    public void saveTagAssociation(String itemId, String tagId) {
```
Comment (jesusmmp) [inline tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/TagServiceImpl.java:91]: I think this save should only do the save... the new and the set should be out of this method

File: vuecomponents/bundle/src/main/bundle/tag-selector_eu.properties [hunk 7]
```diff
@@ -0,0 +1,8 @@
+search_or_add=Buscar o añadir etiqueta
+search_filter=Buscar usando etiquetas
+add_new=Añadir texto como nueva etiqueta
+no_options=No hay etiquetas para el sitio
+no_results=No se han encontrado resultados para la búsqueda actual
+select=Pulsar Enter para seleccionar etiqueta
+selected=Seleccionada
+deselect=Deseleccionar
```
Comment (bgarciaentornos) [inline vuecomponents/bundle/src/main/bundle/tag-selector_eu.properties:8]: @PabloSanRoman would you mind providing translations for EU?

File: vuecomponents/bundle/src/main/bundle/tag-selector_es.properties (hunk 4 above)
Comment (jesusmmp) [reply vuecomponents/bundle/src/main/bundle/tag-selector_es.properties:1]: Great then! :)

File: tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/TagServiceImpl.java (hunk 6 above)
Comment (bgarciaentornos) [reply tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/TagServiceImpl.java:91]: I get what you mean, but I did it this way so the different tools that will be integrated don't have to use TagAssociation objects. They just pass the needed params and that's it, and those lines don't need to be repeated.

Tbh the existing tagService is a bit strange/old and I just started using JPA for my new class, but a refactor could be interesting in the near future...

Comment (jesusmmp) [reply tags/tags-impl/impl/src/java/org/sakaiproject/tags/impl/TagServiceImpl.java:91]: Great! No problem

File: vuecomponents/bundle/src/main/bundle/tag-selector_eu.properties (hunk 7 above)
Comment (PabloSanRoman) [reply vuecomponents/bundle/src/main/bundle/tag-selector_eu.properties:8]: Hi @bgarciaentornos , here it is!

search_or_add=Bilatu edo erantsi etiketa
search_filter=Bilaketa egin etiketak erabiliz
add_new=Gehitu testua etiketa berria gisa
no_options=Etiketarik ez dago gunerako
no_results= Ez da emaitzarik aurkitu bilaketarako
select=Sakatu Enter etiketa aukeratzeko
selected=Hautatua
deselect=Deshautatu

File: assignment/api/src/resources/assignment_mn.properties [hunk 8]
```diff
@@ -603,7 +603,6 @@ uploadall.alert.choose.element=\u0422\u0430 \u0430\u0440\u0445\u0438\u0432\u044B
...
 downloadall.alert.choose.element=\u0422\u0430\u0442\u0430\u0445 \u0445\u0443\u0432\u0438\u043B\u0431\u0430\u0440\u0442 \u0442\u0430 \u0445\u0430\u043C\u0433\u0438\u0439\u043D \u0431\u0430\u0433\u0430\u0434\u0430\u0430 \u043D\u044D\u0433 \u0445\u044D\u0441\u044D\u0433 \u0441\u043E\u043D\u0433\u043E\u0445 \u0451\u0441\u0442\u043E\u0439.
 uploadall.alert.zipFile=\u0422\u0430 \u0437\u0438\u043F \u0430\u0440\u0445\u0438\u0432\u044B\u043D \u0444\u0430\u0439\u043B\u0430\u0430 \u0441\u043E\u043D\u0433\u043E\u043E\u0433\u04AF\u0439 \u0431\u0430\u0439\u043D\u0430. Upload \u0445\u0438\u0439\u0445 \u0444\u0430\u0439\u043B\u0430\u0430\u0440 .zip or a .sit \u0433\u044D\u0441\u044D\u043D \u0430\u0440\u0445\u0438\u0432 \u0444\u0430\u0439\u043B \u0441\u043E\u043D\u0433\u043E\u0445 \u0451\u0441\u0442\u043E\u0439.
-uploadall.size = Upload \u0445\u0438\u0439\u0445 \u0445\u044D\u043C\u0436\u044D\u044D\u043D\u0438\u0439 {0} MB-\u044B\u043D \u0445\u044F\u0437\u0433\u0430\u0430\u0440\u0430\u0430\u0441 \u0445\u044D\u0442\u044D\u0440\u0441\u044D\u043D \u0431\u0430\u0439\u043D\u0430.
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment_mn.properties:606]: Leave these changes out of the PR

File: assignment/api/src/resources/assignment_sv.properties [hunk 9]
```diff
@@ -665,17 +665,13 @@ uploadall.choose.file.gradeFile=Betygsfil (grades.csv-filen i toppniv\u00E5n i d
 uploadall.choose.file.studentSubmissionAttachment=Bilagor till student-inl\u00E4mning(ar)
 uploadall.choose.file.studentSubmissionText=Studentens inl\u00E4mningstext (den inl\u00E4mnade orginaltexten, m\u00F6jligen med l\u00E4rarens inlagda kommentarer)
 uploadall.choose.file2=V\u00E4lj objekt i den komprimerade arkivfilen att ladda upp.
-uploadall.confirm.fileSelect=Du har valt den komprimerade arkivfilen
```
Comment (mpellicer) [inline assignment/api/src/resources/assignment_sv.properties:668]: Leave these changes out of the PR

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 10]
```diff
@@ -3449,6 +3514,10 @@ protected void setAssignmentFormContext(SessionState state, Context context) {
         // get attachment for all purpose object
         putSupplementItemAttachmentStateIntoContext(state, context, ALLPURPOSE_ATTACHMENTS);
 
+        if (serverConfigurationService.getBoolean("tagservice.enable.integrations", false)) {
```
Comment (mpellicer) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:3517]: Must be enabled by default.

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 11]
```diff
@@ -8606,6 +8707,14 @@ private void post_save_assignment(RunData data, String postOrSave) {
                     rubricsService.saveRubricAssociation(RubricsConstants.RBCS_TOOL_ASSIGNMENT, a.getId(), rubricParams);
                 }
 
+                if (serverConfigurationService.getBoolean("tagservice.enable.integrations", false)) {
```
Comment (mpellicer) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:8710]: it must be enabled by default.

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java [hunk 12]
```diff
@@ -14975,57 +15121,6 @@ public void doCancel_options(RunData data, Context context) {
         state.setAttribute(STATE_MODE, MODE_LIST_ASSIGNMENTS);
     } // doCancel_options
 
-    /**
```
Comment (mpellicer) [inline assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:14978]: Is this a cleanup or this logic has been moved?

File: config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties [hunk 13]
```diff
@@ -5492,6 +5492,9 @@
 # The email address that will receive the confirmation mails from the import jobs (by default uses the defined in portal.error.email)
 #tags.import_job_email=youremail@yourdomain.ext
 
+# Allow the use of tags in some tools: assignments, private messages (default is false)
+# tagservice.enable.integrations=true
```
Comment (mpellicer) [inline config/configuration/bundles/src/bundle/org/sakaiproject/config/bundle/default.sakai.properties:5496]: It must be enabled by default, put S2U-32 as prefix for the feature and state that the default is ENABLED

File: master/pom.xml [hunk 14]
```diff
@@ -2046,6 +2046,12 @@
         <version>${sakai.version}</version>
         <scope>provided</scope>
       </dependency>
+      <dependency>
```
Comment (mpellicer) [inline master/pom.xml:2049]: Is this related to this PR? A fix for a different PR?

File: pack/pom.xml [hunk 15]
```diff
@@ -40,6 +40,7 @@
             </activation>
             <properties>
                 <sakai.skip.tests>false</sakai.skip.tests>
+                <sakai.skip.phase>test</sakai.skip.phase>
```
Comment (mpellicer) [inline pack/pom.xml:43]: And this?

File: pack/pom.xml [hunk 16]
```diff
@@ -51,6 +52,7 @@
             <id>skip-tests</id>
             <properties>
                 <sakai.skip.tests>true</sakai.skip.tests>
+                <sakai.skip.phase>none</sakai.skip.phase>
```
Comment (mpellicer) [inline pack/pom.xml:55]: And this?

Comment (bgarciaentornos) [reply pack/pom.xml:55]: This comes from the original vue PR, see https://github.com/sakaiproject/sakai/pull/10484#discussion_r863940978

File: pack/pom.xml (hunk 15 above)
Comment (bgarciaentornos) [reply pack/pom.xml:43]: This comes from the original vue PR https://github.com/sakaiproject/sakai/pull/10484#discussion_r863940978

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java (hunk 12 above)
Comment (bgarciaentornos) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java:14978]: Cleanup, it's code unused for 15 years... I'm putting it back so it can stay for 15 more

File: master/pom.xml (hunk 14 above)
Comment (bgarciaentornos) [reply master/pom.xml:2049]: Belongs to cardgame but I found this issue when modifying the webapi pom.xml

File: assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java (hunk 12 above)
Comment (mpellicer) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: Hahahaha, you're totally right, but let's keep it outside this work.

https://sakaiproject.atlassian.net/browse/SAK-48906

Comment (mpellicer) [reply assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java]: I like this @bgarciaentornos , I'll invite to this one:

https://github.com/sakaiproject/sakai/pull/11574/files

File: master/pom.xml (hunk 14 above)
Comment (mpellicer) [reply master/pom.xml:2049]: Take it outside, i've fixed it!

**Review by mpellicer — APPROVED** [review]

**Comment by mpellicer** [comment]
This is amazing @bgarciaentornos, good job so far!
